File: laba2/lab2.java
import java.util.Scanner;

public class Main {
    public static int searchColor(String r) {
        if (r.length() % 2 != 0) {
            System.out.println("Incorrect value");
            return 0;
        }

        int c = 0;
        int[] vecR = new int[10];
        int[] vecW = new int[10];
        int[] vecB = new int[10];

        for (int i = 0; i < r.length(); i += 2) {
            char color = r.charAt(i);
            int cod = r.charAt(i + 1) - '0';

            switch (color) {
                case 'R': vecR[cod] = 1; break;
                case 'W': vecW[cod] = 1; break;
                case 'B': vecB[cod] = 1; break;
            }
        }

        for (int i = 0; i < 10; i++) {
            if (vecR[i] == 1 && vecW[i] == 1 && vecB[i] == 1) c++;
        }

        return c;
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        System.out.print("Enter rods: ");
        String input = scanner.nextLine();
        System.out.println("Number of rods: " + searchColor(input));
    }
}

File: laba2/lab2.go
package main

import (
	"fmt"
)

func SearchColor(r string) int {
	if len(r)%2 != 0 {
		fmt.Println("Incorrect value")
		return 0
	}

	c := 0
	VecR := make([]int, 10)
	VecW := make([]int, 10)
	VecB := make([]int, 10)

	for i := 0; i < len(r); i += 2 {
		color := r[i]
		cod := int(r[i+1] - '0')

		switch color {
		case 'R':
			VecR[cod] = 1
		case 'W':
			VecW[cod] = 1
		case 'B':
			VecB[cod] = 1
		}
	}

	for i := 0; i < 10; i++ {
		if VecR[i] == 1 && VecW[i] == 1 && VecB[i] == 1 {
			c++
		}
	}
	return c
}

func main() {
	var input string
	fmt.Print("Enter rods: ")
	fmt.Scanln(&input)
	fmt.Println("Number of rods:", SearchColor(input))
}
